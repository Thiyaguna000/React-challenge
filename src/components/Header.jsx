import React, { useEffect, useState } from 'react';
import logo from '../assets/fxdigitallogo.png';

export const Header = () => {
  const [login, setLogin] = useState(false);
  const [user, setUser] = useState('');

  useEffect(() => {
    if (sessionStorage.getItem('login')) {
      setLogin(true);
      setUser(sessionStorage.getItem('user'));
    }
  }, []);

  const Click = () => {
    var navbar = document.querySelector('.main-nav ul');
    navbar.classList.toggle('active');
  };

  const handleLogout = () => {
    sessionStorage.clear();
  };

  return (
    <header>
      <img src={logo} alt="logo" />

      <div href="#" className="toggle-button" onClick={Click}>
        <span className="bar"></span>
        <span className="bar"></span>
        <span className="bar"></span>
      </div>
      {login && (
        <nav className="main-nav">
          <ul>
            <li>
              <a href="/React-challenge/#/home">Home</a>
            </li>
            <li>
              <a href="/React-challenge/#/toprated">Top Rated</a>
            </li>
            <li>
              <a href="/React-challenge/#/upcoming">Upcoming</a>
            </li>
            <li onClick={handleLogout}>
              <a href="/React-challenge/#/login">Logout</a>
            </li>
            <li>
              <span>
                Welcome!<h4>{user}</h4>
              </span>
            </li>
          </ul>
        </nav>
      )}
    </header>
  );
};

export default Header;
