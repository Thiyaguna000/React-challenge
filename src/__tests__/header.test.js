import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

afterEach(cleanup);


it('renders card', () => {
    const tree = renderer
      .create(<Header />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


