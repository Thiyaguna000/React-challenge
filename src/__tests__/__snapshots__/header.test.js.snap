// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card 1`] = `
<header>
  <img
    alt="logo"
    src={
      Object {
        "blurDataURL": "data:image/png;base64,imagedata",
        "height": 24,
        "src": "/img.jpg",
        "width": 24,
      }
    }
  />
  <div
    className="toggle-button"
    href="#"
    onClick={[Function]}
  >
    <span
      className="bar"
    />
    <span
      className="bar"
    />
    <span
      className="bar"
    />
  </div>
</header>
`;
